src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -848,22 +848,53 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
             return sum;
         }
 
-        internal static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(this ImmutableArray<T> items, Func<T, K> keySelector)
+        /// <summary>
+        /// Helper type to give us a PooledDictionary cache in the case where the key for the dictionary is a <see
+        /// cref="ReadOnlyMemory{T}"/>.  These dictionaries need a special comparer, so we cannot use the standard
+        /// pattern of <c>PooledDictionary.GetInstance()</c>.
+        /// </summary>
+        private static class AccumulatorCache<K>
```

Comment:
>AccumulatorCache [](http://example.com/codeflow?start=29&length=16)

I do not think the issue warrants a special type in this file. Also, the name of the type doesn't really "speak" anything specific about its purpose.